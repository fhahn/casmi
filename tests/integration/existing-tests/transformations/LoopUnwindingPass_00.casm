// transform, cmdline "--lu"

CASM LoopUnwindingPass

init foo

rule foo =
{
    forall i in [ 0 .. 2 ] do
    {
        print i
    }
    
    program(self) := undef
}
