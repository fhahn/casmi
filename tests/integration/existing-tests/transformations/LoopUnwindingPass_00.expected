foo=PARBLOCK(STATEMENTS(PARBLOCK(STATEMENTS(PARBLOCK(STATEMENTS(PRINT(LIST(num 0,null),null),null),null),STATEMENTS(PARBLOCK(STATEMENTS(PRINT(LIST(num 1,null),null),null),null),STATEMENTS(PARBLOCK(STATEMENTS(PRINT(LIST(num 2,null),null),null),null),null))),null),STATEMENTS(UPDATE(LOOKUP(ident program,LIST(SELF,null)),UNDEF),null)),null)
